src/lib/registration/widgets/package_search.rb

```diff
@@ -141,151 +139,39 @@ def start_search_event?(event)
       # @param text [String] Text to search for
       def search_package(text)
         return unless valid_search_text?(text)
-        @search = ::Registration::PackageSearch.new(text: text)
         # TRANSLATORS: searching for packages
         Yast::Popup.Feedback(_("Searching..."), _("Searching for packages")) do
-          selected_package_ids = selected_packages.map(&:id)
-          @search.packages.each do |pkg|
-            pkg.select! if selected_package_ids.include?(pkg.id)
-          end
+          controller.search(text)
         end
-        packages_table.change_items(@search.packages)
+        packages_table.change_items(controller.packages)
         update_details
       end
 
       # Finds out the current package which is selected in the packages table
       #
       # @return [RemotePackage,nil]
       def find_current_package
-        return unless search && packages_table.value
-        selected_id = packages_table.value
-        search.packages.find { |p| p.id == selected_id }
+        return unless packages_table.value
+        controller.packages.find { |p| p.id == packages_table.value }
       end
 
       # Selects/unselects the current package for installation
-      #
-      # It does nothing if the package is already installed.
       def toggle_package
         package = find_current_package
-        return if package.nil? || package.installed?
-
-        if package.selected?
-          unselect_package(package)
-        else
-          select_package(package)
-        end
+        return if package.nil?
```

Comment:
this should not happen, not? If toggle is disabled when nothing is selected, it should not be possible to get it.